### PR TITLE
Add ADR for builder updates via dependabot

### PR DESCRIPTION
**Pull request by fwilhe**

### CODE REVIEW

File: docs/architecture/decisions/0003-builder-updates-dependabot.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## Status
+
+Accepted
+
+## Context
+
+The [Garden Linux Builder](https://github.com/gardenlinux/builder) is a containerized solution to build Linux images.
+It is used by the upstream Garden Linux project, but it is and can be used by any project needing custom Linux images.
+
+Linux images are encouraged to pin the builder to a specific build [as shown here in the Garden Linux repository](https://github.com/gardenlinux/gardenlinux/blob/f4a389e88feb15e78b5529831d3cf3cd8978fc20/build#L6).
```
Comment (Vincinator) [inline docs/architecture/decisions/0003-builder-updates-dependabot.md:14]:
```suggestion
All projects that use the builder are encouraged to pin the builder to a specific build [as shown here in the Garden Linux repository](https://github.com/gardenlinux/gardenlinux/blob/f4a389e88feb15e78b5529831d3cf3cd8978fc20/build#L6).
```

File: docs/architecture/decisions/0003-builder-updates-dependabot.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+Pinning the builder to one specific version as opposed to using a `latest` tag is desirable so image builds are stable and predictable, but the current approach has the downside that manual effort is needed to update the used builder from time to time.
```
Comment (Vincinator) [inline docs/architecture/decisions/0003-builder-updates-dependabot.md:15]:
```suggestion
Pinning the builder to one specific version as opposed to using a `latest` tag is desirable so image builds are stable and predictable. This pinning approach has the downside that manual effort is needed to update the used builder from time to time.
```

File: docs/architecture/decisions/0003-builder-updates-dependabot.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+Dependabot can take care to keep this up to date and will open pull requests for updates.
+Those pull requests can validate using CI/CD that the new builder version still works and it will contain a summary of changes between the two versions.
+
+Currently the builder only supports building from Dockerfiles with the standard name (`Dockerfile`), but having a file with that name in the project root just for defining the version of the builder might not be desirable.
+For this reason, the builder should also accept a Dockerfile with a more speaking name such as `builder.dockerfile`.
+Dependabot will be able to pick that up.
+
+## Decision
+
+The Garden Linux builder supports using dependabot and we recommend users to make use of that.
```
Comment (Vincinator) [inline docs/architecture/decisions/0003-builder-updates-dependabot.md:31]:
```suggestion
The Garden Linux builder versions can be bumped with dependabot and we recommend users to make use of that.
```

**Review by Vincinator — COMMENTED** [review]
Looks good so far. Let us sync with @nkraetzschmar before we approve and merge

**Comment by fwilhe** [comment]
@Vincinator @nkraetzschmar updated based on our discussion, hope I got all points we talked about

File: docs/architecture/decisions/0003-builder-updates-dependabot.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+## Decision
+
+The Garden Linux builder versions can be bumped with dependabot and we recommend users to make use of that.
+
+We implement the needed changes to support this, like:
+
+- Tagged releases on GitHub for the builder will be published
+  - The builder implements a versioning schema that is orianted at semantic versioning with a major and a minor version
+    - The minor version will increase on updates that don't break compatibility between the `build` script and the `ghcr.io/gardenlinux/builder` container image
+    - The major versin will increase on updates taht do break compatibility between the `build` script and the `ghcr.io/gardenlinux/builder` container image
```
Comment (Vincinator) [inline docs/architecture/decisions/0003-builder-updates-dependabot.md:45]:
```suggestion
    - The major versin will increase on updates that do break compatibility between the `build` script and the `ghcr.io/gardenlinux/builder` container image
```

**Review by Vincinator — APPROVED** [review]
looks good to me. 

I need to get used to ADR Context chapters. 
It does make sense to write it from the perspective before the ADR is made.

But is feels new to me to describe the context that will be out-dated.
 E.g. "The builder is/has".. 

However, it is completely valid, I just need to get used to it and we need to be consistent.

**Pull request merged**